lib/editor-linter.coffee

```diff
@@ -1,102 +1,42 @@
-{CompositeDisposable, Emitter, Range} = require 'atom'
-Helpers = require './helpers'
+{TextEditor, Emitter, CompositeDisposable} = require('atom')
 
 class EditorLinter
-  constructor: (@linter, @editor) ->
-    @status = true
-    @messages = new Map
-    @inProgress = false
-    @inProgressFly = false
-
-    if @editor is atom.workspace.getActiveTextEditor()
-      @linter.views.updateLineMessages(true)
-
+  constructor: (@editor) ->
+    throw new Error("Given editor isn't really an editor") unless @editor instanceof TextEditor
+    @status = true # Overall linting status
```

Comment:
It's not actually being used anywhere, gonna remove it instead. I have used an alternative approach for `linter:toggle`